extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporter.java

```diff
@@ -167,38 +174,125 @@ public GooglePhotosExporter(GoogleCredentialFactory credentialFactory, JsonFacto
     }
     ContinuationData continuationData = new ContinuationData(nextPageData);
 
-    List<PhotoModel> photos = new ArrayList<>();
-    for (GoogleMediaItem mediaItem : mediaItemSearchResponse.getMediaItems()) {
-      if (mediaItem.getMediaMetadata().getPhoto() != null) {
-        // TODO: address videos later on
-        photos.add(new PhotoModel(
-            "", // TODO: no title?
-            mediaItem.getBaseUrl() + "=d",
-            mediaItem.getDescription(),
-            mediaItem.getMimeType(),
-            mediaItem.getId(),
-            albumId,
-            false));
-      }
+    PhotosContainerResource containerResource = null;
+    GoogleMediaItem[] mediaItems = mediaItemSearchResponse.getMediaItems();
+    if (mediaItems != null && mediaItems.length > 0) {
+      List<PhotoModel> photos = convertPhotosList(albumId, mediaItems, jobId);
+      containerResource = new PhotosContainerResource(null, photos);
     }
 
-    PhotosContainerResource containerResource = new PhotosContainerResource(null, photos);
-
     ResultType resultType = ResultType.CONTINUE;
     if (nextPageData == null) {
       resultType = ResultType.END;
     }
+
     return new ExportResult<>(resultType, containerResource, continuationData);
   }
 
+
+  @VisibleForTesting
+  void populateContainedPhotosList(UUID jobId, TokensAndUrlAuthData authData)
+      throws IOException {
+    // Get list of all album ids
+    List<String> albumIds = new LinkedList<>();
+    AlbumListResponse albumListResponse = getOrCreatePhotosInterface(authData)
+        .listAlbums(Optional.empty());
+    albumIds.addAll(Arrays.stream(albumListResponse.getAlbums()).map(GoogleAlbum::getId)
+        .collect(Collectors.toList()));
+    String token;
+    while ((token = albumListResponse.getNextPageToken()) != null) {
+      albumListResponse = getOrCreatePhotosInterface(authData)
+          .listAlbums(Optional.of(token));
+      albumIds.addAll(
+          Arrays.stream(albumListResponse
+              .getAlbums())
+              .map(GoogleAlbum::getId)
+              .collect(Collectors.toList()));
+    }
+
+    // Get list of all ids belonging to photos belonging to albums
+    TempPhotosData tempPhotosData = jobStore
+        .findData(jobId, createCacheKey(), TempPhotosData.class);
+    if (tempPhotosData == null) {
+      tempPhotosData = new TempPhotosData(jobId);
+      jobStore.create(jobId, createCacheKey(), tempPhotosData);
+    }
+    for (String albumId : albumIds) {
+      MediaItemSearchResponse albumMediaItemResponse = getOrCreatePhotosInterface(authData)
```

Comment:
Is there a reason to re-crete the photos interface every time? If so we should document it.

Reply:
I believe that pattern is to allow for the possibility that auth data might expire.  We haven't written any code to address that yet, however.